testing/opae-cxx/test_buffer_cxx_core.cpp

```diff
@@ -0,0 +1,222 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#define PROTECTION (PROT_READ | PROT_WRITE)
+#ifdef __ia64__
+#define ADDR (void *)(0x8000000000000000UL)
+#define FLAGS_4K (MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED)
+#define FLAGS_2M (FLAGS_4K | MAP_HUGETLB)
+#define FLAGS_1G (FLAGS_2M | MAP_1G_HUGEPAGE)
+#else
+#define ADDR (void *)(0x0UL)
+#define FLAGS_4K (MAP_PRIVATE | MAP_ANONYMOUS)
+#define FLAGS_2M (FLAGS_4K | MAP_HUGETLB)
+#define FLAGS_1G (FLAGS_2M | MAP_1G_HUGEPAGE)
+#endif
+
+#include "test_system.h"
+#include "gtest/gtest.h"
+#include <opae/cxx/core/handle.h>
+#include <opae/cxx/core/properties.h>
+#include <opae/cxx/core/shared_buffer.h>
+#include <opae/cxx/core/token.h>
+#include <sys/mman.h>
+
+using namespace opae::testing;
+using namespace opae::fpga::types;
+
+class buffer_cxx_core : public ::testing::TestWithParam<std::string> {
+protected:
+  buffer_cxx_core() : handle_(nullptr) {}
+
+  virtual void SetUp() override {
+    ASSERT_TRUE(test_platform::exists(GetParam()));
+    platform_ = test_platform::get(GetParam());
+    system_ = test_system::instance();
+    system_->initialize();
+    system_->prepare_syfs(platform_);
+
+    ASSERT_EQ(fpgaInitialize(nullptr), FPGA_OK);
+
+    tokens_ = token::enumerate({properties::get(FPGA_ACCELERATOR)});
+    ASSERT_TRUE(tokens_.size() > 0);
+
+    handle_ = handle::open(tokens_[0], FPGA_OPEN_SHARED);
+    ASSERT_NE(nullptr, handle_.get());
+  }
+
+  virtual void TearDown() override { system_->finalize(); }
+
+  std::vector<token::ptr_t> tokens_;
+  handle::ptr_t handle_;
+  test_platform platform_;
+  test_system *system_;
+};
+
+/**
+ * @test shared_buffer::allocate_null
+ * Calling shared_buffer::allocate with a null handle should throw
+ * invalid_param.
+ */
+TEST_P(buffer_cxx_core, allocate_null) {
+  size_t length = 4096;
+  shared_buffer::ptr_t buf;
+
+  EXPECT_THROW(shared_buffer::allocate(NULL, length), std::exception);
+}
+
+/**
+ * @test shared_buffer::allocate_no_len
+ * Calling shared_buffer::allocate with buffer length = 0 should throw
+ * invalid_param.
+ */
+TEST_P(buffer_cxx_core, allocate_no_len) {
+  size_t length = 4096;
+  shared_buffer::ptr_t buf;
+
+  EXPECT_THROW(shared_buffer::allocate(handle_, 0), std::exception);
```

Comment:
You should be more explicit about the exception

Reply:
In my tests, no other exception type matched the thrown type. Nothing in except.h would match.